### ISSUE TITLE
CSV parsing error with arrow 4.0.0 for evalcast

### ISSUE THREAD

**Issue by eujing** [cmu-delphi/covidcast#534]
Using an example covidhub data file [2021-04-25-Karlen-pypm.csv](https://raw.githubusercontent.com/reichlab/covid19-forecast-hub/master/data-processed/Karlen-pypm/2021-04-25-Karlen-pypm.csv) stored as:
```
.
├── data
│   └── 2021-04-25-Karlen-pypm.csv
└── test.R
```
and snippet `test.R`:
```
library(arrow)
library(tidyverse)

sch <- schema(
            forecast_date=string(),
            target=string(),
            target_end_date=string(),
            location=string(),
            type=string(),
            quantile=string(),
            value=string())

ds = open_dataset("data", format = "csv", schema = sch)

ds %>% select(target) %>% collect()
```

With the latest `arrow==4.0.0` installed, this errors with:
`Error: Invalid: In CSV column #3: CSV conversion error to int64: invalid value 'US'`
However, the snippet runs fine with `arrow==3.0.0`

Notably, `ds %>% collect()` seems to run fine for both versions of arrow and returns a tibble with the right schema, so something about arrow's new implementation of `select` seems to be falling back to inferring the schema and ignoring the specified one.

**Comment by eujing** [comment]
Update, it seems to be [a known bug with arrow 4.0.0](https://issues.apache.org/jira/browse/ARROW-12603#) that has been fixed but is scheduled to release with arrow 5.0.0

**Comment by nmdefries** [comment]
Bug resolved in 4.0.1 release.
